src/Middlewares/RoleOrPermissionMiddleware.php

```diff
@@ -11,20 +11,27 @@ class RoleOrPermissionMiddleware
     public function handle($request, Closure $next, $roleOrPermission, $guard = null)
     {
         $authGuard = Auth::guard($guard);
-        if ($authGuard->guest()) {
-            throw UnauthorizedException::notLoggedIn();
+
+        // For machine-to-machine Passport clients
+        $bearerToken = $request->bearerToken();
+        if ($bearerToken && method_exists($authGuard, 'client')) {
+            $user = $authGuard->client();
         }
 
-        $user = $authGuard->user();
+        $user = $user ?? $authGuard->user();
 
-        if (! method_exists($user, 'hasAnyRole') || ! method_exists($user, 'hasAnyPermission')) {
-            throw UnauthorizedException::missingTraitHasRoles($user);
+        if (! $user) {
+            throw UnauthorizedException::notLoggedIn();
         }
 
         $rolesOrPermissions = is_array($roleOrPermission)
             ? $roleOrPermission
             : explode('|', $roleOrPermission);
 
+        if (! method_exists($user, 'hasAnyRole') || ! method_exists($user, 'hasAnyPermission')) {
+            throw UnauthorizedException::missingTraitHasRoles($user);
+        }
+
```

Comment:
While this is a minor nitpick, moving this up a few lines, before the `$roleOrPermissions = is_array()...` block saves a CPU cycle if this exception needs to be thrown.